google/cloud/spanner_dbapi/parse_utils.py

```diff
@@ -509,11 +518,37 @@ def sql_pyformat_args_to_spanner(sql, params):
             resolved_value = pyfmt % params
             named_args[key] = resolved_value
         else:
+            assert_numeric_precision_and_scale(params[i])
             named_args[key] = params[i]
 
     return sanitize_literals_for_upload(sql), named_args
 
 
+def assert_numeric_precision_and_scale(value):
```

Comment:
I don't know whether this is normal in the Python client, but it seems a little strange that this function is called for __all__ parameter types, and not only numeric values. I would have expected either:
1. That this function is only called for numeric values (i.e. the `if isinstance(value, decimal.Decimal)` would be outside of the function).
2. Or: That there is a generic 'assert_valid_value(value)` function that first checks the type of the input parameter, and then calls the appropriate validation function for that type.

Reply:
makes sense added an instance check before the method is called.